modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/display/context/logic/UIItemsBuilder.java

```diff
@@ -1038,6 +1083,29 @@ else if (_isMoveToTheRecycleBinActionAvailable()) {
 		).build();
 	}
 
+	public DropdownItem createDeleteVersionDropdownItem() {
+		return DropdownItemBuilder.putData(
+			"action", "deleteVersion"
+		).putData(
+			"deleteURL",
+			PortletURLBuilder.create(
+				_getActionURL(
+					"/document_library/edit_file_entry", Constants.DELETE,
+					_getRenderURL(
+						"/document_library/view_file_entry", _getRedirect()
+					).toString())
```

Comment:
This kind of chaining is not allowed by our SF rules. You'll either have to put the result of `_getRenderURL` in a variable and call toString on that, or use String.valueOf(_getRenderURL(...)).